src/main.rs

```diff
@@ -77,19 +78,32 @@ fn main() -> Result<()> {
     let logger = slog::Logger::root(drain, slog::o!());
 
     let config = Config::from_env()?;
-    let path = Path::new(SOCKET_PATH);
+
+    // If we're started using socket activation, use the unix listener at index 0,
+    // else bind manually on SOCKET_PATH.
+    let (listener, listen_address) = match ListenFd::from_env()
+        .take_unix_listener(0)
+        .expect("invalid socket type at index")
```

Comment:
I think it makes slightly more sense to log an error and exit here but I don't feel strongly about it.